src/cephadm/samples/nfs.json

```diff
@@ -0,0 +1,46 @@
+{
+    "pool" : "nfs-ganesha",
+    "namespace" : "nfs-ns",
+    "files": {
+        "ganesha.conf": [
+            "NFS_CORE_PARAM {",
+            "        Enable_NLM = false;",
+            "        Enable_RQUOTA = false;",
+            "        Protocols = 4;",
+            "}",
+            "",
+            "CACHEINODE {",
+            "        Dir_Chunk = 0;",
+            "        NParts = 1;",
+            "        Cache_Size = 1;",
+            "}",
+            "",
+            "EXPORT_DEFAULTS {",
+            "        Attr_Expiration_Time = 0;",
+            "}",
+            "",
+            "NFSv4 {",
+            "        Delegations = false;",
+            "        RecoveryBackend = 'rados_cluster';",
+            "        Minor_Versions = 1, 2;",
+            "}",
+            "",
+            "RADOS_KV {",
+            "        ceph_conf = '/etc/ceph/ceph.conf';",
+            "        userid = admin;",
+            "        nodeid = nfs.a;",
+            "        pool = 'nfs-ganesha';",
+            "        namespace = 'nfs-ns';",
+            "}",
+            "",
+            "RADOS_URLS {",
+            "        ceph_conf = '/etc/ceph/ceph.conf';",
+            "        userid = admin;",
+            "        watch_url = 'rados://nfs-ganesha/nfs-ns/conf-nfs.a';",
+            "}",
+            "",
+            "%url    rados://nfs-ganesha/nfs-ns/conf-nfs.a",
+            ""
+        ]
+    }
+}
```

Comment:
In general, this format looks a bit strange to me. In yaml, this file should IMO look like so:

```yaml
pool: "nfs-ganesha"
namespace: "nfs-ns"
files:
- ganesha.conf: |
    NFS_CORE_PARAM {
     Enable_NLM = false;
     Enable_RQUOTA = false;
     Protocols = 4;
    }
    
    CACHEINODE {
     Dir_Chunk = 0;
     NParts = 1;
     Cache_Size = 1;
    }
    
    EXPORT_DEFAULTS {
     Attr_Expiration_Time = 0;
    }
    
    NFSv4 {
     Delegations = false;
     RecoveryBackend = 'rados_cluster';
     Minor_Versions = 1, 2;
    }
    
    RADOS_KV {
     ceph_conf = '/etc/ceph/ceph.conf';
     userid = admin;
     nodeid = nfs.a;
     pool = 'nfs-ganesha';
     namespace = 'nfs-ns';
    }
    
    RADOS_URLS {
     ceph_conf = '/etc/ceph/ceph.conf';
     userid = admin;
     watch_url = 'rados://nfs-ganesha/nfs-ns/conf-nfs.a';
    }
    
    %url rados://nfs-ganesha/nfs-ns/conf-nfs.a

```

if we convert this to json (as cephadm can't handle json, but `mgr/cephadm` can!), the json could also look like so:


```json
{
  "files": [
    {
      "ganesha.conf": "NFS_CORE_PARAM {\n Enable_NLM = false;\n Enable_RQUOTA = false;\n Protocols = 4;\n}\n\nCACHEINODE {\n Dir_Chunk = 0;\n NParts = 1;\n Cache_Size = 1;\n}\n\nEXPORT_DEFAULTS {\n Attr_Expiration_Time = 0;\n}\n\nNFSv4 {\n Delegations = false;\n RecoveryBackend = 'rados_cluster';\n Minor_Versions = 1, 2;\n}\n\nRADOS_KV {\n ceph_conf = '/etc/ceph/ceph.conf';\n userid = admin;\n nodeid = nfs.a;\n pool = 'nfs-ganesha';\n namespace = 'nfs-ns';\n}\n\nRADOS_URLS {\n ceph_conf = '/etc/ceph/ceph.conf';\n userid = admin;\n watch_url = 'rados://nfs-ganesha/nfs-ns/conf-nfs.a';\n}\n\n%url rados://nfs-ganesha/nfs-ns/conf-nfs.a\n"
    }
  ], 
  "namespace": "nfs-ns", 
  "pool": "nfs-ganesha"
}
```

TBH, I expect people to pass configs via the `ceph orch` parameters, and less via cephadm directly. wdyt?

Reply:
both formats are supported,
```
            if isinstance(config['files'][fname], list):  # type: ignore
                content = '\n'.join(config['files'][fname])  # type: ignore
            else:
                content = config['files'][fname]  # type: ignore
```

Reply:
perfect.